src/media_type.rs

```diff
@@ -36,14 +38,14 @@ pub enum SdpMediaValue {
     Application,
 }
 
-impl ToString for SdpMediaValue {
-    fn to_string(&self) -> String {
-        match *self {
+impl fmt::Display for SdpMediaValue {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let value_string = match *self {
             SdpMediaValue::Audio => "audio",
             SdpMediaValue::Video => "video",
             SdpMediaValue::Application => "application",
-        }
-        .to_string()
+        };
```

Comment:
I don't think there is a reason to ever do `write!(f, "{}", foo) inside of Display::fmt`. You can just do `foo.fmt(f)`. In this case it would be:
```rust
match *self {
    SdpMediaValue::Audio => "audio",
    SdpMediaValue::Video => "video",
    SdpMediaValue::Application => "application",
}.fmt(f)
```